src/wrapper/api_helpers.jl

```diff
@@ -1,59 +1,64 @@
-    # ccall((:Highs_getHighsBoolOptionValue, libhighs), Cint, (Ptr{Cvoid}, Cstring, Ptr{Cint}), highs, option, value)
-# end
-# 
-# function Highs_getHighsIntOptionValue(highs, option, value)
-    # ccall((:Highs_getHighsIntOptionValue, libhighs), Cint, (Ptr{Cvoid}, Cstring, Ptr{Cint}), highs, option, value)
-# end
-# 
-# function Highs_getHighsDoubleOptionValue(highs, option, value)
-    # ccall((:Highs_getHighsDoubleOptionValue, libhighs), Cint, (Ptr{Cvoid}, Cstring, Ptr{Cdouble}), highs, option, value)
-# end
-# 
-# function Highs_getHighsStringOptionValue(highs, option, value)
-    # ccall((:Highs_getHighsStringOptionValue, libhighs), Cint, (Ptr{Cvoid}, Cstring, Cstring), highs, option, value)
-# end
-# 
-# function Highs_getHighsIntInfoValue(highs, info, value)
-    # ccall((:Highs_getHighsIntInfoValue, libhighs), Cint, (Ptr{Cvoid}, Cstring, Ptr{Cint}), highs, info, value)
-# end
-# 
-# function Highs_getHighsDoubleInfoValue(highs, info, value)
-    # ccall((:Highs_getHighsDoubleInfoValue, libhighs), Cint, (Ptr{Cvoid}, Cstring, Ptr{Cdouble}), highs, info, value)
-# end
-# 
-# function Highs_getSolution(highs, colvalue, coldual, rowvalue, rowdual)
-    # ccall((:Highs_getSolution, libhighs), Cvoid, (Ptr{Cvoid}, Ptr{Cdouble}, Ptr{Cdouble}, Ptr{Cdouble}, Ptr{Cdouble}), highs, colvalue, coldual, rowvalue, rowdual)
-# end
-# 
-# function Highs_getBasis(highs, colstatus, rowstatus)
-    # ccall((:Highs_getBasis, libhighs), Cvoid, (Ptr{Cvoid}, Ptr{Cint}, Ptr{Cint}), highs, colstatus, rowstatus)
-# end
 
+"""
+    set_option(highs, option, value::T)
+
+Sets the option to the value, dispatches on
+the value type T to call appropriate HiGHS function.
+"""
+function set_option end
+
+function set_option(highs, option, value::Integer)
+    Highs_setHighsIntOptionValue(highs, option, Cint(value))
+end
+
+function set_option(highs, option, value::Bool)
+    Highs_setHighsBoolOptionValue(highs, option, Cint(value))
+end
+
+function set_option(highs, option, value::AbstractFloat)
+    Highs_setHighsDoubleOptionValue(highs, option, Cdouble(value))
+end
+
+function set_option(highs, option, value::String)
+    Highs_setHighsStringOptionValue(highs, option, value)
+end
+
+"""
+    get_option(highs, option, T)
+
+Gets the option of type T on the HiGHS side.
+"""
+function get_option end
+
+function get_option(highs, option, ::Type{T}) where {T <: Integer}
+    value = Ref{Cint}(0)
+    Highs_getHighsIntOptionValue(highs, option, value)
+    return T(value[])
+end
+
+function get_option(highs, option, ::Type{Bool})
+    value = Ref{Cint}(0)
+    Highs_getHighsBoolOptionValue(highs, option, value)
+    return Bool(value[])
+end
+
+function get_option(highs, option, ::Type{T}) where {T <: AbstractFloat}
+    value = Ref{Cdouble}()
+    Highs_getHighsDoubleOptionValue(highs, option, value)
+    return T(value[])
+end
+
+
+function get_option(highs, option, ::Type{String})
+    p = pointer(String(fill(UInt8(0), 100)))
```

Comment:
I'm not sure this is safe. The GC might collect the `String` object at any time because it's not used later in the Julia code, so the pointer can become invalid. Something like the following should work.
```julia
arr = Array{Cchar}(100)
p = pointer(arr)
...
GC.@preserve arr return unsafe_string(p)
```

Reply:
ok for the pointer creation with Array{Cchar}.

However, the second part does not need the preserve, the string is a copy of the bytes loaded, not a view

```
help?> unsafe_string

  Copy a string from the address of a C-style (NUL-terminated) string encoded
  as UTF-8. (The pointer can be safely freed afterwards.) 
```

Reply:
I think the `GC.@preserve` is needed. The issue wasn't about `unsafe_string` returning a view, it's that `arr` could be freed at any point after the first line, i.e., before the call to `unsafe_string(p)`.

Reply:
Can it be freed before the function ends? The reference to arr is still present through the function so it cannot be GCed before the function ends no?

Reply:
Yes, it can be freed before the function ends. We have to deal with this in other wrappers, e.g., https://github.com/JuliaOpt/SCS.jl/blob/master/src/c_wrapper.jl#L78.

Reply:
got it ok, fixed it in last commit